doc/anaconda.md

```diff
@@ -0,0 +1,70 @@
+Cockpit Storage in Anaconda Mode
+================================
+
+Anaconda (the OS Installer) can open the Cockpit "storaged" page for
+advanced setup of the target storage devices. When this is done,
+storaged is in a special "Anaconda mode" and behaves significantly
+different.
+
+In essence, the storaged page restricts itself to working with the
+target environment. It will hide the real root filesystem (on the USB
+stick that the Live environment was booted from, say), but let the
+user create a "fake" root filesystem on some block device.
+
+Entering Anaconda mode
+----------------------
+
+The "storaged" page is put into Anaconda mode by storing a
+"cockpit_anaconda" item in its `window.localStorage`.  The value
+should be a JSON encoded object, the details of which are explained
+below.
+
+Since both Anaconda and the storaged page are served from the same
+origin, Anaconda can just execute something like this:
+
+```
+    window.localStorage.setItem("cockpit_anaconda",
+                                JSON.stringify({
+                                    "mount_point_prefix": "/sysroot",
+                                    "available_devices": [ "/dev/sda" ]
+                                }));
+    window.open("/cockpit/@localhost/storage/index.html", "storage-tab");
+```
+
+Ignoring storage devices
+------------------------
+
+Anaconda needs to tell Cockpit which devices can not be used to
+install the OS on. This is done with the "available_devices" entry,
+which is an array of strings.
+
+```
+{
+  "available_devices": [ "/dev/sda" ]
+}
```

Comment:
This is a bit misleading or confusing: That sounds like anaconda has to enumerate all available devices, substract the ones which can *not* be used to install, and send the rest? After the paragraph I had expected an `ignore_devices` instead, which might be easier to compute for Anaconda.

Reply:
We started out with "ignore_devices", but I switched it to "available_devices" to make things easier for Anaconda, actually. :-) It already has the list of useable devices, and there is a direct D-Bus call to get them, GetUseableDisks.

Reply:
> After the paragraph I had expected an ignore_devices instead.

Yeah, that's because it was written when we actually had "ignore_devices"...